### PR TITLE
ScientificCalculator and unittests are added

### PR DESCRIPTION
**Pull request by IldiGitHub**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,13 +4,23 @@
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
 
-    <groupId>org.example</groupId>
-    <artifactId>Automation_w1</artifactId>
+    <groupId>com.epam.automation-menthoring</groupId>
+    <artifactId>Calculator</artifactId>
```
Comment (tamasmartin) [inline pom.xml]: package name, artifact id always lowercase, please change it accordingly

File: src/main/java/Calculator/Interface/BaseCalculatorInterface.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package Calculator.Interface;
+
+public interface BaseCalculatorInterface {
+    public double addition(double a, double b);
```
Comment (tamasmartin) [inline src/main/java/Calculator/Interface/BaseCalculatorInterface.java]: interface methods are default public, you can remove the public modifiers from the methods.

File: src/main/java/Main.java [hunk 3]
```diff
@@ -1,10 +1,18 @@
...
     public static void main (String[] args) {
         double a = 3;
         double b = 2;
-        System.out.println("Addition: " + a + " + " + b + " = " + Calculator.addition(a, b));
-        System.out.println("Subtraction: " + a + " - " + b + " = " + Calculator.subtraction(a, b));
-        System.out.println("Multiplication: " + a + " * " + b + " = " + Calculator.multiplication(a, b));
-        System.out.println("Division: " + a + " / " + b + " = " + Calculator.division(a, b));
+
+        Calculator calc = new Calculator();
+        System.out.println("Addition: " + a + " + " + b + " = " + calc.addition(a, b));
```
Comment (tamasmartin) [inline src/main/java/Main.java]: please try String.format instead of + concatenation

File: src/test/java/Calculator/CalculatorTest.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class CalculatorTest {
+
+    @Test
+    public void CalculatorTest(){
+        Calculator calc = new Calculator();
+        double a = 3;
+        double b= 2;
```
Comment (tamasmartin) [inline src/test/java/Calculator/CalculatorTest.java]: please use formatter, your code is not formatted

File: src/test/java/Calculator/ScientificCalculatorTest.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class ScientificCalculatorTest {
+
+    @Test
+    public void ScientificCalculatorTest(){
+        ScientificCalculator scientificCalc = new ScientificCalculator();
+        double a = 6;
+        double b= 3;
```
Comment (tamasmartin) [inline src/test/java/Calculator/ScientificCalculatorTest.java]: please use formatter, your code is not formatted

File: src/main/java/Calculator/Calculator.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
+package Calculator;
```
Comment (tamasmartin) [inline src/main/java/Calculator/Calculator.java]: package name always lowercase

File: src/main/java/Calculator/Calculator.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+import Calculator.Interface.BaseCalculatorInterface;
+
+public class Calculator implements BaseCalculatorInterface {
+
+    public double addition(double a, double b) {
```
Comment (tamasmartin) [inline src/main/java/Calculator/Calculator.java]: We usually add javadoc to public methods. I'll show how: type _/** [Enter]_ before the method.

File: src/main/java/Calculator/ScientificCalculator.java [hunk 8]
```diff
@@ -0,0 +1,11 @@
+package Calculator;
+
+import Calculator.Interface.BaseCalculatorInterface;
```
Comment (tamasmartin) [inline src/main/java/Calculator/ScientificCalculator.java]: This import is unused, please remove.

File: pom.xml [hunk 9]
```diff
@@ -4,13 +4,23 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>org.testng</groupId>
+            <artifactId>testng</artifactId>
+            <version>7.4.0</version>
+            <scope>test</scope>
+        </dependency>
+
+    </dependencies>
+
```
Comment (tamasmartin) [inline pom.xml:59]: We need maven-compiler-plugin to be able to run test from maven.

**Review by tamasmartin — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (IldiGitHub) [reply pom.xml]: Done

File: src/main/java/Calculator/Interface/BaseCalculatorInterface.java (hunk 2 above)
Comment (IldiGitHub) [reply src/main/java/Calculator/Interface/BaseCalculatorInterface.java]: Done

File: src/main/java/Calculator/Calculator.java (hunk 6 above)
Comment (IldiGitHub) [reply src/main/java/Calculator/Calculator.java]: Modified

File: src/main/java/Calculator/Calculator.java (hunk 7 above)
Comment (IldiGitHub) [reply src/main/java/Calculator/Calculator.java]: Added

File: pom.xml (hunk 9 above)
Comment (IldiGitHub) [reply pom.xml:59]: Added, but I was able to run mvn test without it.

File: src/main/java/Main.java (hunk 3 above)
Comment (IldiGitHub) [reply src/main/java/Main.java]: Done

File: src/test/java/Calculator/CalculatorTest.java (hunk 4 above)
Comment (IldiGitHub) [reply src/test/java/Calculator/CalculatorTest.java]: Done, added Save Actions plugin.

File: src/test/java/Calculator/ScientificCalculatorTest.java (hunk 5 above)
Comment (IldiGitHub) [reply src/test/java/Calculator/ScientificCalculatorTest.java]: Done, added Save Actions plugin.

File: src/main/java/Calculator/ScientificCalculator.java (hunk 8 above)
Comment (IldiGitHub) [reply src/main/java/Calculator/ScientificCalculator.java]: Done

**Review by tamasmartin — APPROVED** [review]

**Pull request merged**